app/models/issue_assigner.rb

```diff
@@ -19,28 +19,10 @@ def assign!
   private
 
   def assign_issue_for_sub(sub)
-    issue = Issue.find_by_sql(<<-SQL
-                SELECT
-                  *
-                FROM
-                  issues
-                WHERE
-                  repo_id = '#{sub.repo_id}' and
-                  state   = '#{Issue::OPEN}'
-                  AND id NOT IN (
-                    SELECT
-                      issue_id
-                    FROM
-                      issue_assignments
-                    WHERE
-                      repo_subscription_id = '#{sub.id}'
-                  )
-                ORDER BY
-                  random()
-                LIMIT
-                  1
-                SQL
-                             ).first
+    issue = Issue.where(repo_id: sub.repo_id, state: Issue::OPEN)
+                 .where.not(id: IssueAssignment.select(:issue_id).where(repo_subscription_id: sub.id))
```

Comment:
Instead of `IssueAssignment.select`, which will still generate AR objects, why not go with `pluck`, avoiding AR objects altogether?

Reply:
Makes sense